components/places/src/import/ios_bookmarks.rs

```diff
@@ -0,0 +1,566 @@
+    SELECT structure.child, structure.parent, structure.idx FROM ios.bookmarksBufferStructure structure
+    WHERE EXISTS(
+        SELECT 1 FROM iosBookmarksStaging stage
+        WHERE stage.isLocal = 0
+            AND stage.guid = structure.child
+    );
+REPLACE INTO main.moz_bookmarks_synced_structure(guid, parentGuid, position)
+    SELECT structure.child, structure.parent, structure.idx FROM ios.bookmarksLocalStructure structure
+    WHERE EXISTS(
+        SELECT 1 FROM iosBookmarksStaging stage
+        WHERE stage.isLocal != 0
+            AND stage.guid = structure.child
+    );
+";
+
+lazy_static::lazy_static! {
+    static ref POPULATE_STAGING: String = format!(
+        "INSERT OR IGNORE INTO temp.iosBookmarksStaging(
+            guid,
+            type,
+            parentid,
+            pos,
+            title,
+            bmkUri,
+            keyword,
+            tags,
+            date_added,
+            modified,
+            isLocal
+        )
+        SELECT
+            b.guid,
+            b.type,
+            b.parentid,
+            b.pos,
+            b.title,
+            CASE
+                WHEN b.bmkUri IS NOT NULL
+                    THEN validate_url(b.bmkUri)
+                ELSE NULL
+            END,
+            b.keyword,
+            b.tags,
+            sanitize_timestamp(b.date_added),
+            sanitize_timestamp(b.server_modified),
+            0
+        FROM ios.bookmarksBuffer b
+        WHERE NOT b.is_deleted
+            -- Skip anything also in `local` (we can't use `replace`,
+            -- since we use `IGNORE` to avoid inserting bad records)
+            AND (
+                (b.guid IN {roots})
+                OR
+                (b.guid NOT IN (SELECT l.guid FROM ios.bookmarksLocal l))
+            )
+        ;
+        INSERT OR IGNORE INTO temp.iosBookmarksStaging(
+            guid,
+            type,
+            parentid,
+            pos,
+            title,
+            bmkUri,
+            keyword,
+            tags,
+            date_added,
+            modified,
+            isLocal
+        )
+        SELECT
+            l.guid,
+            l.type,
+            l.parentid,
+            l.pos,
+            l.title,
+            l.bmkUri,
+            l.keyword,
+            l.tags,
+            sanitize_timestamp(l.date_added),
+            sanitize_timestamp(l.local_modified),
+            1
+        FROM ios.bookmarksLocal l
+        WHERE NOT l.is_deleted
+        ;",
+        roots = ROOTS,
+    );
+
+
+    static ref CREATE_STAGING_TABLE: String = format!("
+        CREATE TEMP TABLE temp.iosBookmarksStaging(
+            id INTEGER PRIMARY KEY,
+            guid TEXT NOT NULL UNIQUE
+                CHECK(length(guid) == 12),
+            type TINYINT NOT NULL
+                CHECK(type == {ios_bookmark_type} OR type == {ios_folder_type} OR type == {ios_separator_type}),
+            parentid TEXT,
+            pos INT,
```

Comment:
We use it to populate moz_bookmarks_synced.pos. It's only set for separators, I think (unlike idx in the structure).

Reply:
Marking as resolved. We could consider removing moz_bookmarks_synced.pos if we don't actually need it, but that would be separate. As it is, it seems reasonable to populate it, since it doesn't seem useless.

Reply:
Uh, `moz_bookmarks_synced` doesn't have a `pos`?